unittests/resources/config/settings.py

```diff
@@ -194,6 +193,27 @@ def hostname():
                     'environs': ['builtin']
                 }
             ]
+        },
+        {
+            'name': 'sys3',
+            'descr': 'System required for testing default extras',
+            'hostnames': [r'sys3\d+'],
+            'partitions': [
+                {
+                    'name': 'part1',
+                    'descr': 'Login nodes',
+                    'scheduler': 'local',
+                    'launcher': 'local',
+                    'environs': ['builtin']
+                },
+                {
+                    'name': 'part2',
+                    'descr': 'Compute nodes',
+                    'scheduler': 'slurm',
+                    'launcher': 'srun',
+                    'environs': ['builtin']
+                }
+            ]
```

Comment:
I don't think you need a new system definition. You could use the existing ones in the unit tests.

Reply:
The one above that has a slurm scheduler has already `extras` so it does not trigger the bug.